crates/rattler_conda_types/src/match_spec/parse.rs

```diff
@@ -370,6 +375,26 @@ impl NamelessMatchSpec {
     }
 }
 
+fn parse_url(input: &str) -> Result<Url, ParseMatchSpecError> {
+    match Url::parse(input) {
+        Ok(url) => Ok(url),
+        Err(_) => {
+            #[cfg(not(target_arch = "wasm32"))]
+            {
+                match PathBuf::from_str(input) {
```

Comment:
This is platform dependent. Therefore it's better to use the `file_url` crate.

Reply:
The problem is that I can't use file_url to start with as it will break if it isn't an URL. So I start with checking for a path, which succeeds for both and then map the file_url::file-path_to_url if it is a path. otherwise it will try to parse as an normal url.

Reply:
Resolved!